### DISCUSSION TITLE
Custom json request body parsing and swagger/redoc generation

### DISCUSSION THREAD

**Discussion by maiorfi** [fastapi/fastapi#7104] [Questions] [answered]
Hi.

I have an endpoint whose intended use is that of having an "index" query path parameter and a simple bool payload, that cannot be treated as a pydantic model, so I implemented it as follows:

```python
@app.post("/output/{index}")
async def write_output(index: int, request: Request):
    requestedState: bool = await request.json()

    logger.debug(f'write_output({index},{requestedState})...')

    # ....
```

Is there a way to let bool payload appear in swagger/redoc?

Thanks!

**Comment by Mause** [comment]
Have you tried using a pydantic model and setting `__root__` to bool?

**Comment by maiorfi** [comment, marked as answer]
Do you mean something like that?

```python
class OutputPayload(BaseModel):
    __root__: bool


@app.post("/output/{index}")
async def write_output(index: int, outputPayload: OutputPayload):
    requestedState: bool = outputPayload.__root__

    logger.debug(f'write_output({index},{requestedState})...')
    # ...
```

**Comment by tiangolo** [comment]
Thanks for the help @Mause ! :cake: 

I assume you solved it, thanks for closing the issue @maiorfi  :+1:
